spec/features/users/sign_up_spec.rb

```diff
@@ -23,6 +23,20 @@
     end
   end
 
+  context 'user cancels on the enter password screen' do
+    before do
+      email = 'test@test.com'
+      visit sign_up_email_path
+      submit_form_with_valid_email
+      click_confirmation_link_in_email(email)
+    end
+
+    it 'returns them to the home page' do
+      click_on t('links.cancel_account_creation')
+      expect(current_path).to eq root_path
```

Comment:
Can we also make sure the user has been deleted?
```ruby
user = User.find_with_email(email)
expect(user).to be_nil
```

I would remove the `before` block and put everything in the `it` block. That way, you have access to the `email` variable without having to turn it into an instance variable.